lib/ocamlorg_web/graphql.ml

```diff
@@ -0,0 +1,307 @@
+open Ocamlorg
+
+type package_dep =
+  { name : string
+  ; constraints : string option
+  }
+
+type packages_result =
+  { total_packages : int
+  ; packages : Package.t list
+  }
+
+let starts_with s1 s2 =
+  let len1 = String.length s1 in
+  if len1 > String.length s2 then
+    false
+  else
+    let s1 = String.lowercase_ascii s1 in
+    let s2 = String.lowercase_ascii s2 in
+    String.(equal (sub s2 0 (length s1)) s1)
+
+let is_package s1 s2 =
+  let len1 = String.length s1 in
+  if len1 <> String.length s2 then
+    false
+  else
+    let s1 = String.lowercase_ascii s1 in
+    let s2 = String.lowercase_ascii s2 in
+    String.(equal s2 s1)
+
+let get_packages_result total_packages offset limit filter packages =
+  if filter = "sortByName" then
+    let packages =
+      List.filteri (fun i _ -> offset <= i && i <= offset + limit) packages
```

Comment:
I think we want `offset <= i && i < offset + limit` here as we are off by one (asking for `offset: 10, limit: 10` returns `11` packages, I would suspect most would think it should return 10) ? This would be a good thing to test in the follow-up PR.